File: AspNetMvcUnleashed/0672329980_source/Chapter21/CS/UnleashedBlog/UnleashedBlog/Helpers/BlogLinkHelper.cs
using System.Web.Mvc;
using System.Web.Mvc.Html;
using UnleashedBlog.Models;

namespace UnleashedBlog.Helpers
{
    public static class BlogLinkHelper
    {
        public static string BlogLink(this HtmlHelper helper, BlogEntry entry)
        {
            return helper.RouteLink(entry.Title, "Details", new { year = entry.DatePublished.Year, month = entry.DatePublished.Month, day = entry.DatePublished.Day, name = entry.Name });
        }
    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter21/VB/UnleashedBlog/Helpers/BlogLinkHelper.vb

Namespace Helpers

    Public Module BlogLinkHelper

        <System.Runtime.CompilerServices.Extension()> _
        Public Function BlogLink(ByVal helper As HtmlHelper, ByVal entry As BlogEntry) As String
            Return helper.RouteLink(entry.Title, "Details", New With {Key .year = entry.DatePublished.Year, Key .month = entry.DatePublished.Month, Key .day = entry.DatePublished.Day, Key .name = entry.Name})
        End Function
    End Module

End Namespace
